Las/Utils/Authorization.cs

```diff
@@ -0,0 +1,194 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+
+using Lucidtech.Las.Core;
+
+namespace Lucidtech.Las.Utils
+{
+    public class AmazonAuthorization
+    {
+        /* General information */
+        private string Region { get; }
+        private string Service { get; }
+
+        /* Keys the customer receives from Lucidtech */
+        private string AwsApiKey { get; }
+        private string AwsAccessKey { get; }
+        private string AwsSecretKey { get; }
+        
+        private const string Algorithm = "AWS4-HMAC-SHA256";
+
+        public AmazonAuthorization(Credentials credentials)
+        {
+            AwsApiKey = credentials.ApiKey ;
+            AwsSecretKey = credentials.SecretAccessKey;
+            AwsAccessKey = credentials.AccessKeyId;
+            Region = "eu-west-1";
+            Service = "execute-api";
+        }
+
+        public Dictionary<string, string> SignHeaders(Uri uri, string method, byte [] body)
+        {
+            if (body == null)
+            {
+                body = Encoding.UTF8.GetBytes("");
+            }
+            
+            string amzDate = AmzDate();
+            string dateStamp = Datestamp();
+
+            Dictionary<string, string> headers = Headers(uri, amzDate);
+            
+            byte [] canonicalRequest = GetCanonicalRequest(uri, method, body, headers);
+            
+            string reqDigest = ComputeHashSha256(canonicalRequest);
+            string credScope = GetCredentialScope(dateStamp);
+            byte[] stringToSign = GetSignString(amzDate, credScope, reqDigest);
+            byte[] signingKey = GetSignatureKey(dateStamp);
+            string signature = StringFromByteArray(SignHash(signingKey, stringToSign));
+                 
+            var authHeader = BuildAuthHeader(
+                amzDate: amzDate,
+                signature: signature,
+                credScope:credScope,
+                signedHeaders:headers.Keys.ToList()
+            );
+            return authHeader;
+        }
+        
+        /* Public Static Methods */
+        public static string StringFromByteArray(byte[] bytes)
```

Comment:
Is there no built-in functionality for this?

Reply:
Not that I could find...